### PR TITLE
Support externally specified examples.

### PR DESCRIPTION
**Pull request by marina-p**
A new option, 'ExampleConfigFilePath', allows specifying example payloads
and maintaining them outside of the Swagger/OpenAPI specification.

Testing:
- added new tests, manual testing

### CODE REVIEW

**Review by PatGod — CHANGES_REQUESTED** [review]

File: docs/user-guide/CompilerConfig.md [hunk 1]
```diff
@@ -52,10 +52,12 @@
...
+* *ExampleConfigFilePath* is the path to the file containing metadata about example parameter payloads.  If it is not specified, and *DiscoverExamples* is set to ```false```, the compiler looks for a default file named ```examples.json``` in the specified examples directory.  If *DiscoverExamples* is false, every time an example is used in the Swagger file, RESTler will first look for it in metadata, and if found this will override the payload in the spec.
 
+* *ExamplesDirectory* is the directory where the compiler will copy example payloads found in the Swagger file if *DiscoverExamples* (see below) is true.  This allows first discovering, then updating the examples to set your own values, and maintaining them separately from the Swagger file.
```
[the remark this answers is not in the document]
Comment (marina-p) [reply docs/user-guide/CompilerConfig.md:58]: Thanks, great points, let me work on this a bit more.

[the remark this answers is not in the document]
Comment (marina-p) [reply docs/user-guide/CompilerConfig.md:58]: Let me know if the new version addresses these comments.

---
In reply to: [613671896](https://github.com/microsoft/restler-fuzzer/pull/186#discussion_r613671896) [](ancestors = 613671896)

**Review by PatGod — DISMISSED** [review]

**Comment by mbiuki** [comment]
Are the above changes also documented? Is `Examples.md` going to be updated to reflect the above?
Could we test with creating new example files if I update RESTler? I am seeking some sample example files to follow. Thanks!

**Comment by marina-p** [comment]
@mbiuku Yes, documented in the change.  Please let us know if you try this out and have suggestions for improving the doc.

**Pull request merged**